security/cmd/flexvolume/driver/driver.go

```diff
@@ -0,0 +1,511 @@
+	// on the node.
+	NodeAgentWorkloadHomeDir string `json:"nodeagent_workload_home,omitempty"`
+	// Relative location to NodeAgentManagementHomeDir where per workload credential
+	// files will be created.
+	// Default: /creds
+	// For example: /creds here implies /var/run/nodeagent/creds/ directory
+	NodeAgentCredentialsHomeDir string `json:"nodeagent_credentials_home,omitempty"`
+	// Whether node agent will be notified of workload using gRPC or by creating files in
+	// NodeAgentCredentialsHomeDir.
+	// Default: false
+	UseGrpc bool `json:"use_grpc,omitempty"`
+	// If UseGrpc is true then host the UDS socket here.
+	// This is relative to NodeAgentManagementHomeDir
+	// Default: mgmt.sock
+	// For example: /mgmt/mgmt.sock implies /var/run/nodeagement/mgmt/mgmt.sock
+	NodeAgentManagementAPI string `json:"nodeagent_management_api,omitempty"`
+	// Log level for loggint to node syslog. Options: INFO|WARNING
+	// Default: WARNING
+	LogLevel string `json:"log_level,omitempty"`
+}
+
+// FlexVolumeInputs is the structure used by kubelet to notify driver
+// volume mounts/unmounts.
+type FlexVolumeInputs struct {
+	UID            string `json:"kubernetes.io/pod.uid"`
+	Name           string `json:"kubernetes.io/pod.name"`
+	Namespace      string `json:"kubernetes.io/pod.namespace"`
+	ServiceAccount string `json:"kubernetes.io/serviceAccount.name"`
+}
+
+const (
+	versionK8s         string = "1.8"
+	configFileName     string = "/etc/flexvolume/nodeagent.json"
+	nodeAgentHome      string = "/var/run/nodeagent"
+	mountDir           string = "/mount"
+	credentialDirHome  string = "/creds"
+	managementSockHome string = "/mgmt.sock"
+	logLevelWarn       string = "WARNING"
+)
+
+var (
+	// configuration is the active configuration that is being used by the driver.
+	configuration *ConfigurationOptions
+	// logWriter is used to notify syslog of the functionality of the driver.
+	logWriter *syslog.Writer
+	// defaultConfiguration is the default configuration for the driver.
+	defaultConfiguration = ConfigurationOptions{
+		K8sVersion:                  versionK8s,
+		NodeAgentManagementHomeDir:  nodeAgentHome,
+		NodeAgentWorkloadHomeDir:    mountDir,
+		NodeAgentCredentialsHomeDir: credentialDirHome,
+		UseGrpc:                     false,
+		NodeAgentManagementAPI:      managementSockHome,
+		LogLevel:                    logLevelWarn,
+	}
+	configFile = configFileName
+	// setup as var's so that we can test.
+	getExecCmd = exec.Command
+)
+
+// InitCommand handles the init command for the driver.
+func InitCommand() error {
+	if configuration.K8sVersion == "1.8" {
+		resp, err := json.Marshal(&InitResponse{Status: "Success", Message: "Init ok.", Attach: false})
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(resp))
+		return nil
+	}
+	return genericSuccess("init", "", "Init ok.")
+}
+
+// checkValidMountOpts checks if there are sufficient inputs to
+// call node agent.
+func checkValidMountOpts(opts string) (*pb.WorkloadInfo, bool) {
```

Comment:
Instead of returning bool, you can just return the error.

Reply:
This function converts the options string to WorkloadInfo proto message. Can you change the function name to something like produceWorkloadInfo?